frontend/gateway/pb/gateway.proto

```diff
@@ -162,3 +166,55 @@ message PongResponse{
 	repeated moby.buildkit.v1.apicaps.APICap LLBCaps = 2 [(gogoproto.nullable) = false];
 	repeated moby.buildkit.v1.types.WorkerRecord Workers = 3;
 }
+
+message NewContainerRequest {
+	string ContainerID = 1;
+	// For mount input values we can use random identifiers passed with ref
+	repeated pb.Mount Mounts = 2;
+	pb.NetMode Network = 3;
+	pb.SecurityMode Security = 4;
+}
+
+message NewContainerResponse{}
+
+message ReleaseContainerRequest {
+	string ContainerID = 1;
+}
+
+message ReleaseContainerResponse{}
+
+message ExecMessage {
+	string ProcessID = 1;
+	oneof Input {
+		InitMessage Init = 2;
+		FdMessage File = 3;
+		ResizeMessage Resize = 4;
+		StartedMessage Started = 5;
+		ExitMessage Exit = 6;
+	}
+}
+
+message InitMessage{
+	string ContainerID = 1;
+	pb.Meta Meta = 2;
+	repeated uint32 Fds = 3;
+	bool Tty = 4;
+}
+
+message ExitMessage {
+	uint32 Code = 2;
```

Comment:
Why start with 2?

Reply:
hah, whoops, refactored before opening pr and missed this, will fix.